### ISSUE TITLE
Site Editor: Remove the Navigation Menus Sidebar

### ISSUE THREAD

**Issue by Copons** [Automattic/wp-calypso#71278]
The "Navigation Menus" sidebar is in a rough beta state and is more confusing than helpful.

<img width="367" alt="Screenshot 2022-12-16 at 17 39 38" src="https://user-images.githubusercontent.com/2070010/208156868-0e6751a5-0263-473f-96e2-acf3a7162d76.png">

We should remove it for the time being. It will likely show up again in a better state somewhere else in the Gutenberg UI (for example in the dark left sidebar).

_Originally requested in p1670010686600519-slack-C048CUFRGFQ and pc4f5j-2Fb-p2#comment-3228_

**Comment by Copons** [comment]
This looks much more complicated than anticipated, as the Navigation Menus sidebar is not registered via hook, but it depends on the `process.env.IS_GUTENBERG_PLUGIN` feature flag (which is inherently enabled on Dotcom and, as you can imagine, is not something we can disable haphazardly).

https://github.com/WordPress/gutenberg/blob/c05d7c5535a38ae18661c8c7e99c1aec72720722/packages/edit-site/src/components/sidebar-edit-mode/index.js#L67-L75

Furthermore, the Navigation Menus is connected to the Navigation block's "Open list view" button (currently re-labelled "Open navigation list view" in the current Gutenberg trunk).

<img width="336" alt="Screenshot 2022-12-21 at 14 16 44" src="https://user-images.githubusercontent.com/2070010/208926235-ceaa0b9b-781a-40c3-a22e-cfb51e746b68.png">

The functionality is undergoing radical changes with the off-canvas navigation editor experiment (`__experimentalEnableOffCanvasNavigationEditor`) that, if I understand it correctly, should open the navigation menu in the block's setting sidebar.

https://github.com/WordPress/gutenberg/pull/46335

Given that, in this instance, the feature flag changed from plugin to experimental:

```diff
- if ( process.env.IS_GUTENBERG_PLUGIN )
+ if ( window?.__experimentalEnableOffCanvasNavigationEditor === true )
```

I have the feeling that the Navigation Menus will be either removed altogether or gated by a different feature flag, because otherwise I'm not quite sure what is the point of it anymore.

@scruffian I see that you are closely involved with the new approach. Do you have any insider intel about the fate of the Navigation Menus sidebar?

**Comment by scruffian** [comment]
What we could do is just hide this feature behind the `window?.__experimentalEnableOffCanvasNavigationEditor`. I created a PR for this here: https://github.com/WordPress/gutenberg/pull/46710

**Comment by Copons** [comment]
@scruffian nice! I wasn't sure if the two features were related or if the inspector view would make the nav sidebar obsolete. Thanks for opening the PR! 🙇‍♂️

**Comment by Copons** [comment]
https://github.com/WordPress/gutenberg/pull/46710 has been merged and should be included in 14.9. Dotcom can wait. 🙂

**Comment by Copons** [comment]
Circling back, this has been fixed a while ago when Gutenberg 14.9 was merged on Dotcom! 🎉 
Thanks again @scruffian!
